TwilioVerifyDemo/TwilioVerifyDemo/FactorDetail/Presenter/FactorDetailPresenter.swift

```diff
@@ -51,6 +55,18 @@ class FactorDetailPresenter {
 }
 
 extension FactorDetailPresenter: FactorDetailPresentable {
+  var isSilentyApproved: Bool {
+    AppModel.factorsSilentyApproved[factor.sid] == true
+  }
+  
+  func updateSilentlyApproveStatus() {
+    if !AppModel.factorsSilentyApproved.keys.contains(factor.sid) {
+      AppModel.factorsSilentyApproved[factor.sid] = false
+    }
+    
+    AppModel.factorsSilentyApproved[factor.sid]?.toggle()
```

Comment:
Does this logic override it self? Maybe we can use `guards` here to establish unique paths for both scenarios.

Reply:
Yes and no, it is indeed overriding itself, but that is because the dictionary could be empty at the first time, so in the line 64 we assign the value to an entry given by the `factor.sid` key and then we switch the value. The next time this method gets called, it will only switch the value rather than assign it